app/models/device.rb

```diff
@@ -0,0 +1,76 @@
+class Device < ApplicationRecord
+  belongs_to :user
+
+  validates :token, uniqueness: { scope: %i[user_id platform app_bundle] }
+
+  def create_notification(title, body, payload)
+    case platform
+    when "iOS"
+      ios_notification(title, body, payload)
+    when "Android"
+      android_notification(title, body, payload)
+    end
+  end
+
+  private
+
+  def ios_notification(title, body, payload)
+    n = Rpush::Apns2::Notification.new
+    # [@forem/oss] `.where().first` is necessary because we use Redis data storage
+    # https://github.com/rpush/rpush/wiki/Using-Redis#find_by_name-cannot-be-used-in-rpush-redis
+    # rubocop:disable Rails/FindBy
+    n.app = Rpush::Apns2::App.where(name: app_bundle).first || recreate_ios_app
+    # rubocop:enable Rails/FindBy
+
+    n.device_token = token
+    n.data = {
+      aps: {
+        :alert => {
+          title: ApplicationConfig["COMMUNITY_NAME"],
+          subtitle: title,
+          body: body
+        },
+        "thread-id" => ApplicationConfig["COMMUNITY_NAME"]
+      },
+      data: payload
+    }
+    n.save!
+  end
+
+  def android_notification(title, body, payload)
+    n = Rpush::Gcm::Notification.new
+    # [@forem/oss] `.where().first` is necessary because we use Redis data storage
+    # https://github.com/rpush/rpush/wiki/Using-Redis#find_by_name-cannot-be-used-in-rpush-redis
+    # rubocop:disable Rails/FindBy
+    n.app = Rpush::Gcm::App.where(name: app_bundle).first || recreate_android_app
+    # rubocop:enable Rails/FindBy
+
+    n.registration_ids = [token]
+    n.priority = "high"
+    n.content_available = true
+    n.notification = { title: title, body: body }
+    n.data = { data: payload }
+    n.save!
+  end
+
+  def recreate_ios_app
```

Comment:
I'd maybe follow Ruby-conventions and make this and the next method bang methods (i.e. `recreate_ios_app!`) as they are "destructive".

Reply:
✅